index.js

```diff
@@ -1483,14 +1489,101 @@ class Hydra extends EventEmitter {
   }
 
   /**
-   * @name _getRandomServicePort
-   * @summary Retrieves a random TCP/IP port.
-   * @return {number} port - new random socket port
+   * @name _parseServicePortConfig
+   * @summary Parse and process given port data in config
+   * @param {mixed} port - configured port
+   * @return {promise} promise - resolving with unassigned port, rejecting when no free port is found
    */
-  _getRandomServicePort() {
-    const maxSocketPort = 65535;
-    const nonPriviliagePortBountry = 1024;
-    return parseInt(nonPriviliagePortBountry + (new Date().getTime() % (Math.random() * (maxSocketPort - nonPriviliagePortBountry))));
+  _parseServicePortConfig(port) {
+    //No port given, get unassigned port from standard ranges
+    if (typeof port === 'undefined' || !port || port == 0) {
+      port = '1024-65535';
+    }
+    return new Promise((resolve, reject) => {
+      var portRanges = port.toString().split(",")
+        .map( p => {
+          p = p.trim();
+          const ipRe = '(102[4-9]|10[3-9]\\d|1[1-9]\\d{2}|[2-9]\\d{3}|[1-5]\\d{4}|6[0-4]\\d{3}|65[0-4]\\d{2}|655[0-2]\\d|6553[0-5])';
+          let matches = p.match(new RegExp(`^${ipRe}-${ipRe}$`, 'g'));
+          if (matches !== null) {
+            return p;
+          } else {
+            matches = p.match(new RegExp(`^${ipRe}$`, 'g'));
+            if (matches !== null) {
+              return p;
+            }
+          }
+          return null;
+        })
+        .filter(p => p != null);
+      let receivedCallBacks = 0;
+      if (portRanges.length == 0) {
+        reject('servicePort configuration does not contain no valid port(s)');
```

Comment:
Grammatical improvement: servicePort configuration does not contain valid port(s)